Library/ViewModels/SettingsNewslettersCellViewModel.swift

```diff
@@ -0,0 +1,181 @@
+import KsApi
+import Prelude
+import ReactiveSwift
+import Result
+
+public protocol SettingsNewslettersCellViewModelInputs {
+
+  func allNewslettersSwitchTapped(on: Bool)
+  func awakeFromNib()
+  func configureWith(value: (newsletter: Newsletter, user: User))
+  func newslettersSwitchTapped(on: Bool)
+}
+
+public protocol SettingsNewslettersCellViewModelOutputs {
+
+  var showOptInPrompt: Signal<String, NoError> { get }
+  var subscribeToAllSwitchIsOn: Signal<Bool?, NoError> { get }
+  var switchIsOn: Signal<Bool?, NoError> { get }
+  var unableToSaveError: Signal<String, NoError> { get }
+  var updateCurrentUser: Signal<User, NoError> { get }
+}
+
+public protocol SettingsNewslettersCellViewModelType {
+
+  var inputs: SettingsNewslettersCellViewModelInputs { get }
+  var outputs: SettingsNewslettersCellViewModelOutputs { get }
+}
+
+public final class SettingsNewsletterCellViewModel: SettingsNewslettersCellViewModelType,
+SettingsNewslettersCellViewModelInputs, SettingsNewslettersCellViewModelOutputs {
+
+  public init() {
+
+    let newsletter = self.newsletterProperty.signal.skipNil()
+
+    let initialUser = self.awakeFromNibProperty.signal
+      .flatMap {
+        AppEnvironment.current.apiService.fetchUserSelf()
```

Comment:
Do we still need to fetch the user since we're now passing in through the `configure` function? I think you can prob just use `initialUserProperty`

Reply:
Oh, good call. I forgot to delete this. 👍